### PR TITLE
Employment loan fix

### LINKED ISSUE #874: [Examples] - Polish and needed checks
# Apps - Small Review

[protocol-monorepo/examples/employment-based-loan/contracts at dev · superfluid-finance/protocol-monorepo · GitHub](https://github.com/superfluid-finance/protocol-monorepo/tree/dev/examples/employment-based-loan/contracts)

- No checking if caller is host or with the right agreement.
- If not needed don’t save CFA and host if using CFALib (already point this in older PR: https://github.com/superfluid-finance/protocol-monorepo/pull/772)

[protocol-monorepo/BudgetNFT.sol at dev · superfluid-finance/protocol-monorepo · GitHub](https://github.com/superfluid-finance/protocol-monorepo/blob/dev/examples/budget-nft/contracts/BudgetNFT.sol)

- Not using CFALib
- Why Migrations.sol file?

[protocol-monorepo/examples/nftbillboard-userdata at dev · superfluid-finance/protocol-monorepo · GitHub](https://github.com/superfluid-finance/protocol-monorepo/tree/dev/examples/nftbillboard-userdata)
- Remove .github folder
- If not needed don’t save CFA and host if using CFALib (already talked about this in older PR: https://github.com/superfluid-finance/protocol-monorepo/pull/772)

[protocol-monorepo/DividendRightsToken.sol at dev · superfluid-finance/protocol-monorepo · GitHub](https://github.com/superfluid-finance/protocol-monorepo/blob/dev/examples/rewards-distribution-token/contracts/DividendRightsToken.sol)

- No checking if caller is host or with the right agreement.
- If not needed don’t save HOST if using IDALib

### PR DESCRIPTION
**Pull request by saflamini**
cc @ngmachado 

1) Modifiers added which check that the host and CFA are calling the callbacks
2) Host and CFA state vars removed - using the CFAv1 to get these now
3) Cleaned up several small comments, removed unecessary imports in the mockv3aggregator file
4) Made small change to test file to reflect that we are no longer checking that the host + cfa state vars are correct

### CODE REVIEW

File: examples/employment-based-loan/contracts/EmploymentLoan.sol [hunk 1]
```diff
@@ -384,8 +385,34 @@ contract EmploymentLoan is SuperAppBase {
...
+    ) 
+        external 
+        override 
+        onlyHost
+        returns (bytes memory newCtx) 
     {
-        newCtx = _updateOutflow(ctx);
+        if (!_isCFAv1(_agreementClass)) {
+            newCtx = ctx;
+            return newCtx;
```
Comment (ngmachado) [inline examples/employment-based-loan/contracts/EmploymentLoan.sol]: return ctx directly

File: examples/employment-based-loan/contracts/EmploymentLoan.sol [hunk 2]
```diff
@@ -384,8 +385,34 @@ contract EmploymentLoan is SuperAppBase {
...
+        }
+        return _updateOutflow(ctx);
+    }
+
+    function _isCFAv1(address agreementClass) private view returns (bool) {
+        bytes32 CFA_ID = keccak256("org.superfluid-finance.agreements.ConstantFlowAgreement.v1");
```
Comment (ngmachado) [inline examples/employment-based-loan/contracts/EmploymentLoan.sol]: consider if makes sense having CFA_ID as constant variable, and not compute every time there is callbacks.

File: examples/employment-based-loan/contracts/EmploymentLoan.sol [hunk 3]
```diff
@@ -48,10 +45,9 @@ contract EmploymentLoan is SuperAppBase {
         AggregatorV3Interface _priceFeed,
         uint8 _priceFeedDecimals
     ) {
-        host = _host;
-        cfa = IConstantFlowAgreementV1(
+        IConstantFlowAgreementV1 cfa = IConstantFlowAgreementV1(
             address(
-                host.getAgreementClass(
+                _host.getAgreementClass(
                     keccak256(
```
Comment (ngmachado) [inline examples/employment-based-loan/contracts/EmploymentLoan.sol]: If CFA_ID is constant this should change

Comment (jtriley2p) [reply examples/employment-based-loan/contracts/EmploymentLoan.sol]: Can confirm this is constant. It is always `v1`, even in testing environments.

Comment (saflamini) [reply examples/employment-based-loan/contracts/EmploymentLoan.sol]: @ngmachado @JoshuaTrujillo15 see most recent commit

CFA_ID has been added as a constant variable and the above syntax has been simplified

File: examples/employment-based-loan/contracts/EmploymentLoan.sol (hunk 2 above)
Comment (saflamini) [reply examples/employment-based-loan/contracts/EmploymentLoan.sol]: Added as constant variable

File: examples/employment-based-loan/contracts/EmploymentLoan.sol (hunk 1 above)
Comment (saflamini) [reply examples/employment-based-loan/contracts/EmploymentLoan.sol]: done - removed duplication

**Review by ngmachado — APPROVED** [review]
LGTM - thank you for the fast fix! 🥳

**Pull request merged**